### PR TITLE
Tweak unhideAll to change visibility of all inputs or outputs

### PR DESCRIPTION
**Pull request by alpaix**
The purpose of this PR is to allow `unhideAll` action to alter visibility of all cell inputs and outputs independently. 

Key changes are:
- `inputHidden` and `outputHidden` attributes will be optional
- `unhideAll` reducer will test attributes values to be not `undefined` before applying the change

Resoves #4665

### CODE REVIEW

File: packages/actions/src/actions/cells.ts [hunk 1]
```diff
@@ -196,11 +196,9 @@ export function updateCellExecutionCount(payload: {
...
 }
 
-export function unhideAll(payload: {
-  outputHidden: boolean;
-  inputHidden: boolean;
-  contentRef: ContentRef;
-}): actionTypes.UnhideAll {
+export function unhideAll(
+  payload: actionTypes.UnhideAll["payload"]
+): actionTypes.UnhideAll {
```
Comment (captainsafia) [inline packages/actions/src/actions/cells.ts:201]: Thanks for making this incremental clean-up!

File: packages/reducers/src/core/entities/contents/notebook.ts [hunk 2]
```diff
@@ -655,15 +655,19 @@ function unhideAll(
   state: NotebookModel,
   action: actionTypes.UnhideAll
 ): RecordOf<DocumentRecordProps> {
+  const metadataMixin: any = {};
```
Comment (captainsafia) [inline packages/reducers/src/core/entities/contents/notebook.ts]: Nit: Can we add a strict type here? It'll probably be helpful as we move to resolve the to-do in this code to ensure that we are parsing input/output hidden properties from different front-ends correctly.

**Review by captainsafia — APPROVED** [review]
Looks good! Left one comment in-line that would be good to resolve before merging.

File: packages/reducers/src/core/entities/contents/notebook.ts (hunk 2 above)
Comment (alpaix) [reply packages/reducers/src/core/entities/contents/notebook.ts]: Done.

**Pull request merged**